src/__tests__/rest/compliance/fungible/base.ts

```diff
@@ -0,0 +1,151 @@
+import { assertTagPresent } from '~/assertions';
+import { TestFactory } from '~/helpers';
+import { RestClient } from '~/rest';
+import { createAssetParams } from '~/rest/assets/params';
+import { ProcessMode, TxBase } from '~/rest/common';
+import {
+  bothConditionsRequirements,
+  complianceRequirementParams,
+  complianceRequirementsParams,
+  receiverConditionsRequirements,
+  senderConditionsRequirements,
+} from '~/rest/compliance';
+import { Identity } from '~/rest/identities/interfaces';
+
+const handles = ['issuer', 'blocked', 'investor'];
+let factory: TestFactory;
+
+describe('Compliance Requirements for Fungible Assets', () => {
+  let restClient: RestClient;
+  let signer: string;
+  let issuer: Identity;
+  let blocked: Identity;
+  let investor: Identity;
+  let ticker: string;
+  let signerTxBase: TxBase;
+
+  beforeAll(async () => {
+    factory = await TestFactory.create({ handles });
+    ({ restClient } = factory);
+    issuer = factory.getSignerIdentity(handles[0]);
+    blocked = factory.getSignerIdentity(handles[1]);
+    investor = factory.getSignerIdentity(handles[2]);
+
+    ticker = factory.nextTicker();
+    signer = issuer.signer;
+    signerTxBase = { options: { signer, processMode: ProcessMode.Submit } };
+
+    await restClient.assets.createAsset(createAssetParams(ticker, signerTxBase));
+  });
+
+  afterAll(async () => {
+    await factory.close();
+  });
+
+  it('method: getComplianceRequirements', async () => {
+    const result = await restClient.compliance.getComplianceRequirements(ticker);
+
+    expect(result).toMatchObject({
+      requirements: expect.any(Array),
+    });
+  });
+
+  it('method: setRequirements', async () => {
+    const params = complianceRequirementsParams(
+      [
+        bothConditionsRequirements(issuer.did, ticker, blocked.did, 'Us'),
+        senderConditionsRequirements(issuer.did),
+        receiverConditionsRequirements(investor.did),
+      ],
+      signerTxBase
+    );
+    const txData = await restClient.compliance.setRequirements(ticker, params);
+
+    expect(txData).toEqual(assertTagPresent(expect, 'complianceManager.replaceAssetCompliance'));
+
+    const requirements = await restClient.compliance.getComplianceRequirements(ticker);
+
+    expect(requirements).toMatchObject({
+      requirements: expect.arrayContaining([
+        expect.objectContaining({ conditions: params.requirements[0] }),
+      ]),
+    });
+  });
+
+  it('method: pauseRequirements', async () => {
+    const txData = await restClient.compliance.pauseRequirements(ticker, signerTxBase);
+
+    expect(txData).toEqual(assertTagPresent(expect, 'complianceManager.pauseAssetCompliance'));
```

Comment:
isPaused should be checked here, and after unpaused instead of having that case at the bottom